airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -501,6 +540,79 @@ def read_chunked(
             yield from super().read(getter, params)
 
 
+class CRMSearchStream(IncrementalStream, ABC):
+
+    limit = 100  # This value is used only when state is None.
+    state_pk = "updatedAt"
+    updated_at_field = "updatedAt"
+    
+    @property
+    def url(self):
+        if self.state:
+            return f"/crm/v3/objects/{self.entity}/search"
+        return f"/crm/v3/objects/{self.entity}"
+
+    def __init__(
+        self,
+        entity: Optional[str] = None,
+        last_modified_field: Optional[str] = None,
+        associations: Optional[List[str]] = None,
+        include_archived_only: bool = False, **kwargs
+    ):
+        super().__init__(**kwargs)
+        self._state = None
+        self.entity = entity or self.entity
+        self.last_modified_field = last_modified_field or self.last_modified_field
+        self.associations = associations or self.associations
+        self._include_archived_only = include_archived_only
+
+    def list(self, fields) -> Iterable:
```

Comment:
I know other classes are already defining a `list` method but this is a reserved python word.

Reply:
Moreover, this function looks like the one defined in `CRMObjectStream`. Do you think you could make them share a parent class?

Reply:
The `list` function is actually being called when the connector's `read` command is invoked. I think it's called internally, so even though it does not look good because it's a reserved word, I don't know if we can rename it.

Reply:
I found where it was being invoked, so I renamed it in 42b1f42. I'm sorry I didn't know much about how these methods connected with the framework.